jmix-flowui/flowui/src/main/java/io/jmix/flowui/component/grid/headerfilter/DataGridHeaderFilter.java

```diff
@@ -0,0 +1,383 @@
+        overlay.addClassName(COLUMN_FILTER_DIALOG_CLASSNAME);
+
+        if (!isSmallDevice()) {
+            overlay.addClassName(COLUMN_FILTER_POPUP_CLASSNAME);
+        } else {
+            overlay.addClassName(COLUMN_FILTER_FOOTER_SMALL_CLASSNAME);
+        }
+
+        overlay.getFooter().add(createApplyButton(), createCancelButton());
+        overlay.addOpenedChangeListener(this::onOpenOverlay);
+        overlay.addDialogCloseActionListener(__ -> doCancel());
+    }
+
+    protected void initPropertyFilter() {
+        propertyFilter = uiComponents.create(PropertyFilter.class);
+
+        propertyFilter.setDataLoader(context.getDataLoader());
+        propertyFilter.setProperty(context.getProperty());
+        propertyFilter.setAutoApply(false);
+
+        appliedOperation = propertyFilterSupport.getDefaultOperation(
+                context.getMetaClass(), context.getProperty());
+        propertyFilter.setOperation(appliedOperation);
+        propertyFilter.setOperationEditable(true);
+        propertyFilter.setParameterName(PropertyConditionUtils.generateParameterName(context.getProperty()));
+        propertyFilter.setWidthFull();
+    }
+
+    protected void onOpenOverlay(Dialog.OpenedChangeEvent event) {
+        if (event.isOpened()) {
+            propertyFilter.focus();
+        }
+    }
+
+    protected JmixButton createApplyButton() {
+        JmixButton applyButton = uiComponents.create(JmixButton.class);
+        applyButton.setIcon(VaadinIcon.CHECK.create());
+        applyButton.setText(messages.getMessage("columnFilter.apply.text"));
+
+        applyButton.addThemeVariants(ButtonVariant.LUMO_PRIMARY);
+        applyButton.addClickListener(__ -> {
+            doApply();
+            overlay.close();
+
+            fireEvent(new ApplyEvent(this, propertyFilter, false));
+        });
+
+        if (isSmallDevice()) {
+            applyButton.getStyle().set("flex-grow", "1");
+        }
+
+        return applyButton;
+    }
+
+    protected JmixButton createCancelButton() {
+        JmixButton cancelButton = uiComponents.create(JmixButton.class);
+        cancelButton.setIcon(VaadinIcon.BAN.create());
+        cancelButton.setText(messages.getMessage("columnFilter.cancel.text"));
+
+        cancelButton.addClickListener(__ -> doCancel());
+
+        if (isSmallDevice()) {
+            cancelButton.getStyle().set("flex-grow", "1");
+        }
+
+        return cancelButton;
+    }
+
+    /**
+     * Applies the current value of the {@link PropertyFilter} to the loader
+     */
+    public void doApply() {
+        propertyFilter.getDataLoader().load();
+        appliedValue = propertyFilter.getValue();
+        appliedOperation = propertyFilter.getOperation();
+
+        filterButton.getElement().setAttribute(COLUMN_FILTER_BUTTON_ACTIVATED_ATTRIBUTE_NAME,
+                propertyFilter.getValue() != null);
+    }
+
+    @SuppressWarnings({"unchecked"})
+    protected void doCancel() {
```

Comment:
no public API for `cancel`?

Reply:
`Cancel` resets the values that were applied before the overlay-dialog was opened.
It makes no sense to open the overlay programmatically, scince you have access to the property filter directly.

A valid use case for programmatic usage - get the proeprty filter -> change the values -> apply it.

So, no public API for `Cancel`.